docs/layers/eks/design-decisions/decide-on-secrets-management-for-eks.md

```diff
@@ -0,0 +1,62 @@
+
+Use [External Secrets Operator](https://external-secrets.io/latest/) with AWS SSM Parameter Store.
+
+External Secrets Operator is a Kubernetes operator that manages and stores sensitive information in external secret management systems like AWS Secrets Manager, GCP Secret Manager, Azure Key Vault, HashiCorp Vault, and more. It allows you to use these external secret management systems to securely add secrets in your Kubernetes cluster.
+
+Cloud Posse historically recommends using External Secrets Operator with AWS SSM Parameter Store and has existing Terraform modules to support this solution. See the [eks/external-secrets-operator](/components/library/aws/eks/external-secrets-operator/) component.
+
+### Option 2: AWS Secrets Manager secrets with Kubernetes Secrets Store CSI Driver
+
+Use [AWS Secrets and Configuration Provider (ASCP) for the Kubernetes Secrets Store CSI Driver](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver.html). This option allows you to use AWS Secrets Manager secrets as Kubernetes secrets that can be accessed by Pods as environment variables or files mounted in the pods. The ASCP also works with [Parameter Store parameters](https://docs.aws.amazon.com/systems-manager/latest/userguide/integrating_csi_driver.html)
+
+However, Cloud Posse does not have existing Terraform modules for this solution. We would need to build this support.
+
+### Option 3: SOPS Operator
+
+Use [SOPS Operator](https://github.com/isindir/sops-secrets-operator) to manage secrets in Kubernetes. SOPS Operator is a Kubernetes operator that builds on the `sops` project by Mozilla to encrypt the sensitive portions of a `Secrets` manifest into a `SopsSecret` resource, and then decrypt and provision `Secrets` in the Kubernetes cluster.
+
+1. **Mozilla SOPS Encryption**: Mozilla SOPS (Secrets OPerationS) is a tool that encrypts Kubernetes secret manifests, allowing them to be stored securely in Git repositories. SOPS supports encryption using a variety of key management services. Most importantly, it supports AWS KMS which enables IAM capabilities for native integration with AWS.
+
+2. **GitOps-Compatible Secret Management**: In a GitOps setup, secrets are typically stored in version-controlled repositories. However, storing plain-text secrets in Git is a security risk. With SOPS, sensitive information in Kubernetes secret manifests is encrypted, so only the encrypted version is stored in the Git repo. This ensures that the secrets are kept secure while still benefiting from GitOps practices like version control, auditability, and CI/CD pipelines.
+
+3. **AWS KMS Integration**: If AWS KMS is used, SOPS encrypts the secrets using customer-managed keys (CMKs) in AWS KMS. This ensures that only those with appropriate permissions can decrypt and use the secrets. The encrypted secret manifests can be committed to the Git repository, while the keys used for encryption are securely managed in AWS.
```

Comment:
```suggestion
3. **AWS KMS Integration**: SOPS uses AWS KMS to encrypt secrets with customer-managed keys (CMKs), ensuring only authorized users—based on IAM policies—can decrypt them. The encrypted secret manifests can be safely committed to Git, with AWS securely managing the keys. Since it's IAM-based, it integrates seamlessly with STS tokens, allowing secrets to be decrypted inside the cluster without hardcoded credentials.
```